plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/extractor/JenkinsfileVisitor.java

```diff
@@ -36,67 +43,198 @@ public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations v
         return variableDeclarations;
     }
 
+    @Override
+    public J visitMapEntry(G.MapEntry mapEntry, PluginMetadata pluginMetadata) {
+        super.visitMapEntry(mapEntry, pluginMetadata);
+        J.MethodInvocation method = getCursor().firstEnclosing(J.MethodInvocation.class);
+        if (method == null) {
+            return mapEntry;
+        }
+        if (method.getSimpleName().equals("buildPlugin")) {
+            if ("useContainerAgent".equals(mapEntry.getKey().toString())) {
+                if (mapEntry.getValue() instanceof J.Literal literal) {
+                    pluginMetadata.setUseContainerAgent(Boolean.valueOf(literal.getValueSource()));
+                } else if (mapEntry.getValue() instanceof J.Identifier) {
+                    Expression resolvedArg = resolveVariable(mapEntry.getValue());
+                    if (resolvedArg instanceof J.Literal literal) {
+                        pluginMetadata.setUseContainerAgent(Boolean.valueOf(literal.getValueSource()));
+                    }
+                }
+            }
+            if ("forkCount".equals(mapEntry.getKey().toString())) {
+                if (mapEntry.getValue() instanceof J.Literal literal) {
+                    pluginMetadata.setForkCount(literal.getValue().toString());
+                } else if (mapEntry.getValue() instanceof J.Identifier) {
+                    Expression resolvedArg = resolveVariable(mapEntry.getValue());
+                    if (resolvedArg instanceof J.Literal literal) {
+                        pluginMetadata.setForkCount(literal.getValue().toString());
+                    }
+                }
+            }
+        }
+        return mapEntry;
+    }
+
     @Override
     public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, PluginMetadata pluginMetadata) {
         LOG.debug("Visiting method invocation {}", method);
         method = super.visitMethodInvocation(method, pluginMetadata);
         if ("buildPlugin".equals(method.getSimpleName())) {
             List<Expression> args = method.getArguments();
 
-            List<Integer> jdkVersions =
-                    args.stream().flatMap(this::extractJdkVersions).distinct().toList();
+            // Empty args means Java 8 in Windows and Linux
+            if (args.size() == 1 && args.get(0) instanceof J.Empty) {
+                pluginMetadata.addPlatform(new PlatformConfig(Platform.LINUX, JDK.JAVA_8, null, true));
+                pluginMetadata.addPlatform(new PlatformConfig(Platform.WINDOWS, JDK.JAVA_8, null, true));
+                return method;
+            }
 
-            LOG.info("JDK versions found: {}", jdkVersions);
+            List<PlatformConfig> platforms = new LinkedList<>();
```

Comment:
I'm really not proud of this code.

We should use different visit callback and use cursor to get information higher on the tree. Not going down

But I least now I have test to validation behavior and will improve in the future

See `visitMapEntry` that only visit entries inside a method

```java
J.MethodInvocation method = getCursor().firstEnclosing(J.MethodInvocation.class);
``